### PR TITLE
[ADMIN] Add Political Functions to ElectedRepresentative

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

**Comment by illusionOfParadise** [comment]
[ch10958]

File: src/Entity/ElectedRepresentative/PoliticalFunction.php [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    }
+
+    public function setElectedRepresentative(ElectedRepresentative $electedRepresentative): void
+    {
+        $this->electedRepresentative = $electedRepresentative;
+    }
+
+    public function __toString(): string
+    {
+        return array_search($this->name, PoliticalFunctionNameEnum::CHOICES);
```
Comment (ottaviano) [inline src/Entity/ElectedRepresentative/PoliticalFunction.php]: `(string) array_search(...)` as the function can return `false|int|string` types

File: src/Form/MandateType.php [hunk 2]
```diff
@@ -41,22 +37,22 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
-                'label' => 'Périmètre géographique',
             ])
+            ->add('geographicalArea', TextType::class)
         ;
 
         $builder->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $event) {
             $data = $event->getData();
 
-            if (!isset($data['isElected']) || false === $data['isElected']) {
+            if (!isset($data['isElected']) || false === $data['isElected']
```
Comment (ottaviano) [inline src/Form/MandateType.php]: What about removing this listener entirely as you have added 2 `@Assert\Expression` on `finishAt` field and JS code for disable/enable the same field?

File: templates/form_theme_admin.html.twig [hunk 3]
```diff
@@ -131,38 +111,140 @@
...
+            });
+        });
+    </script>
 {% endblock %}
 
 {% block app_admin_elected_representative_mandate_mandates_mandates_entry_widget -%}
-    <div class="row" style="margin-top: 10px; margin-left: -30px;">
-        <div class="col-md-2">
+    <div class="row mandate" style="margin-top: 10px; margin-left: -30px;">
+        <div class="col-md-2{{ form.type.vars.valid ? '' : ' error' }}">
```
Comment (ottaviano) [inline templates/form_theme_admin.html.twig]:
```twig
<div class="col-md-2{{ form.type.vars.valid ? '' : ' error' }}">
```
as shorter alternative =>
```twig
<div class="col-md-2 {{ not form.type.vars.valid ? 'error' }}">
```

File: src/Entity/ElectedRepresentative/PoliticalFunction.php [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+{
+    /**
+     * @ORM\Id
+     * @ORM\Column(type="integer")
+     * @ORM\GeneratedValue
+     */
+    private $id;
+
+    /**
+     * @var string
```
Comment (Remg) [inline src/Entity/ElectedRepresentative/PoliticalFunction.php]: `string|null`

File: src/Entity/ElectedRepresentative/PoliticalFunction.php [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+     * @var string|null
+     *
+     * @ORM\Column(length=255, nullable=true)
+     *
+     * @Assert\Length(max="255")
+     */
+    private $clarification;
+
+    /**
+     * @var string
```
Comment (Remg) [inline src/Entity/ElectedRepresentative/PoliticalFunction.php]: `string|null`

File: src/Entity/ElectedRepresentative/PoliticalFunction.php [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+
+    /**
+     * @var bool
+     *
+     * @ORM\Column(type="boolean", options={"default": true})
+     */
+    private $onGoing = true;
+
+    /**
+     * @var \DateTime
```
Comment (Remg) [inline src/Entity/ElectedRepresentative/PoliticalFunction.php]: `\DateTime|null`

File: src/Form/PoliticalFunctionType.php [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+            ->add('onGoing', CheckboxType::class, [
+                'label' => false,
+                'required' => false,
+            ])
+            ->add('beginAt', 'sonata_type_date_picker')
+            ->add('finishAt', 'sonata_type_date_picker', [
+                'required' => false,
+            ])
+            ->add('geographicalArea', TextType::class, [
+            ])
```
Comment (Remg) [inline src/Form/PoliticalFunctionType.php]: you can remove the third parameter?

**Pull request merged**